java/src/main/java/ai/rapids/cudf/MemoryBuffer.java

```diff
@@ -146,6 +146,39 @@ public final long getAddress() {
     return address;
   }
 
+  /**
+   * Copy a subset of src to this buffer starting at destOffset using the specified CUDA stream.
+   * The copy has completed when this returns, but the memory copy could overlap with
+   * operations occurring on other streams.
+   * @param destOffset the offset in this to start copying from.
+   * @param src what to copy from
+   * @param srcOffset offset into src to start out
+   * @param length how many bytes to copy
+   * @param stream CUDA stream to use
+   */
+  public final void copyFromMemoryBuffer(
+          long destOffset, MemoryBuffer src, long srcOffset, long length, Cuda.Stream stream) {
+    addressOutOfBoundsCheck(address + destOffset, length, "copy range dest");
+    src.addressOutOfBoundsCheck(src.address + srcOffset, length, "copy range src");
+    Cuda.memcpy(address + destOffset, src.address + srcOffset, length, CudaMemcpyKind.DEFAULT, stream);
```

Comment:
Should the 'CudaMemcpyKind` be `DEVICE_TO_HOST`

Reply:
No this needs to be `DEFAULT` since it's not a requirement that this is a host buffer and `src` is a device buffer.